containers/Connector/Connector.tsx

```diff
@@ -160,13 +163,31 @@ const useConnector = () => {
 		// eslint-disable-next-line react-hooks/exhaustive-deps
 	}, [isAppReady]);
 
+	useEffect(() => {
+		if (signer) {
+			let provider,
+				account: any = signer.getAddress();
+			const infuraUrl = 'https://mainnet.infura.io/v3/';
+
+			provider = new ethers.providers.JsonRpcProvider(
+				infuraUrl + process.env.NEXT_PUBLIC_INFURA_PROJECT_ID
+			);
+
+			getENSNameAndAvatarUrl(account, provider).then((ensObj) => {
+				if (ensObj !== null) {
+					setEns(ensObj?.name);
+				}
+			});
+		}
+	}, [signer]);
+
 	// load previously saved wallet
 	useEffect(() => {
 		// disables caching if in IFrame allow parent to set wallet address
-		if (onboard && selectedWallet && !walletAddress && !isIFrame()) {
+		if (onboard && selectedWallet && !walletAddress && !ensName && !ensAvatar && !isIFrame()) {
```

Comment:
Not sure why this logic was added/needed?

Reply:
I didn't add that logic in, so im not sure either
Felt i should leave it alone